unit-tests/unit-tests-live.cpp

```diff
@@ -5915,3 +5917,156 @@ TEST_CASE("l500_presets_set_preset", "[live]")
+                {
+                    if (!(sp.stream_type() == profile.stream && sp.fps() == fps && sp.stream_index() == profile.index && sp.format() == profile.format)) continue;
+                    auto vid = sp.as<rs2::video_stream_profile>();
+                    auto h = vid.height();
+                    auto w = vid.width();
+                    auto format = vid.format(); // RS2_FORMAT_YUYV ?
+                    if (!(w == profile.width && h == profile.height)) continue;
+                    if (sp.stream_type() == RS2_STREAM_COLOR)
+                        rgb_stream_profile.push_back(sp);
+                    if (sp.stream_type() == RS2_STREAM_INFRARED)
+                        ir_stream_profile.push_back(sp);
+                }
+            }
+        }
+        typedef enum configuration
+        {
+            IR_ONLY,
+            IR_RGB,
+            IR_RGB_EXPOSURE
+        }configuration;
+
+        configuration tests[3] = { IR_ONLY, IR_RGB, IR_RGB_EXPOSURE };
+        std::map< configuration, std::map<int, std::vector<unsigned long long>>> frames_num_info;
+        std::map < configuration, rs2_metadata_type> actual_fps;
+        std::map<int, std::vector<unsigned long long>> curr_frames_num_info;
+        rs2_metadata_type curr_actual_fps = 0;
+        std::mutex mutex;
+        std::map<int, std::string> stream_names;
+
+        auto process_frame = [&](const rs2::frame& f)
+        {
+            auto stream_type = f.get_profile().stream_name();
+            auto frame_num = f.get_frame_number();
+            auto unique_id = f.get_profile().unique_id();
+            if (std::find(curr_frames_num_info[unique_id].begin(), curr_frames_num_info[unique_id].end(), frame_num) != curr_frames_num_info[unique_id].end()) // check if frame is already processed
+                return;
+            curr_frames_num_info[unique_id].push_back(frame_num);
+            if (std::string(stream_type) == "Color")
+                return;
+            if(!curr_actual_fps)
+                curr_actual_fps = f.get_frame_metadata(RS2_FRAME_METADATA_ACTUAL_FPS);
+        };
+
+        auto frame_callback = [&](const rs2::frame& f)
+        {
+            std::lock_guard<std::mutex> lock(mutex);
+            if (rs2::frameset fs = f.as<rs2::frameset>())
+            {
+                // With callbacks, all synchronized stream will arrive in a single frameset
+                for (const rs2::frame& ff : fs)
+                {
+                    process_frame(ff);
+                }
+            }
+            else
+            {
+                // Stream that bypass synchronization (such as IMU) will produce single frames
+                process_frame(f);
+            }
+        };
+        rgb_sensor.close();
+        ir_sensor.close();
+        for (auto& test : tests)
+        {
+            curr_actual_fps = 0;
+
+            if (test == IR_RGB_EXPOSURE)
+                set_exposure(list, 90000); // set exposure value to x > 1000/fps
```

Comment:
Adjust scenario to 60 (start)->30 (actual) fps change.
Replace list with the actual sensor